cdap-ui/app/features/adapters/services/AdapterCreateModel.js

```diff
@@ -0,0 +1,172 @@
+angular.module(PKG.name + '.feature.adapters')
+  .factory('AdapterCreateModel', function(AdapterApiFactory, $state, $timeout, $q, mySettings) {
+    var defaultSource = {
+      name: 'Add a source',
+      properties: {},
+      placeHolder: true
+    };
+    var defaultSink = {
+      name: 'Add a sink',
+      properties: {},
+      placeHolder: true
+    };
+    var defaultTransforms = [{
+      name: 'Add a transform',
+      properties: {},
+      placeHolder: true
+    }];
+
+    var defaultSchedule = {
+      cron: '* * * * *'
+    };
+
+    function Model () {
+      var defaultMetadata = {
+          name: '',
+          description: '',
+          type: 'ETLBatch'
+      };
+      this.metadata= angular.copy(defaultMetadata);
+      this.resetPlugins();
+    }
+
+    Model.prototype.resetPlugins = function resetPlugins() {
+      this.source = angular.copy(defaultSource);
+      this.transforms = angular.copy(defaultTransforms);
+      this.sink = angular.copy(defaultSink);
+      this.schedule = angular.copy(defaultSchedule);
+    };
+
+    Model.prototype.setMetadata = function (metadata) {
+      // FIXME: There is a timing issue when editing a draft.
+      this.metadata.type = metadata.type;
+      this.metadata.name = metadata.name;
+      this.metadata.description = metadata.description;
+    };
+
+    Model.prototype.setSource = function setSource(source) {
+      this.source = angular.copy(source);
+    };
+
+    Model.prototype.setTransform = function setTransform(transform) {
+      if (!transform || !transform.length) {
+        return;
+      }
+
+      if (this.transforms[0].placeHolder) {
+        this.transforms = transform.slice();
+      } else {
+        this.transforms = this.transforms.concat(transform);
+      }
+    };
+
+    Model.prototype.setSink = function setSink(sink) {
+      this.sink = sink;
+    };
+
+    Model.prototype.setSchedule = function setSchedule(schedule) {
+      this.schedule = schedule || defaultSchedule;
+    };
+
+    Model.prototype.saveAsDraft = function saveAsDraft() {
+      var defer = $q.defer();
+      if (!this.metadata.name.length) {
+        defer.reject({
+          message: 'Please provide a name for the Adapter to be saved as draft'
+        });
+        return defer.promise;
+      }
+      var adapterDrafts = {};
+      this.adapterDrafts[this.metadata.name] = {
+        config: {
+          metadata: this.metadata,
+          source: this.source,
+          transforms: this.transforms,
+          sink: this.sink,
+          schedule: this.schedule
+        }
+      };
+
+      return mySettings.set('adapterDrafts', this.adapterDrafts);
+    }
+
+    Model.prototype.save = function save() {
+      var defer = $q.defer();
+      if (this.source.placeHolder && this.sink.placeHolder) {
+        defer.reject({
+          message: 'Adapter needs atleast a source/sink'
```

Comment:
I think it should be `source and a sink`, instead of `source/sink`.
How it currently is might give the idea that only one of them is required.